src/Microsoft.AspNetCore.Mvc.Razor/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.Razor.Host;
+using Microsoft.AspNetCore.Mvc.ViewComponents;
+using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
+using Microsoft.AspNetCore.Razor.Runtime.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor
+{
+    /// <summary>
+    /// Provides methods to create tag helper representations of view components.
+    /// </summary>
+    public class ViewComponentTagHelperDescriptorFactory
+    {
+        private readonly IViewComponentDescriptorProvider _descriptorProvider;
+
+        /// <summary>
+        /// Creates a new <see cref="ViewComponentTagHelperDescriptorFactory"/>, 
+        /// then creates <see cref="TagHelperDescriptor"/>s for <see cref="ViewComponent"/>s 
+        /// in the given <see cref="IViewComponentDescriptorProvider"/>. 
+        /// </summary>
+        /// <param name="descriptorProvider">The provider of <see cref="ViewComponentDescriptor"/>s.</param>
+        public ViewComponentTagHelperDescriptorFactory(IViewComponentDescriptorProvider descriptorProvider)
+        {
+            if (descriptorProvider == null)
+            {
+                throw new ArgumentNullException(nameof(descriptorProvider));
+            }
+
+            _descriptorProvider = descriptorProvider;
+        }
+
+        /// <summary>
+        /// Creates <see cref="TagHelperDescriptor"/> representations of <see cref="ViewComponent"/>s
+        /// in an <see href="Assembly"/> represented by the given <paramref name="assemblyName"/>.
+        /// </summary>
+        /// <param name="assemblyName">The name of the assembly containing 
+        /// the <see cref="ViewComponent"/>s to translate.</param>
+        /// <returns>A <see cref="IEnumerable{TagHelperDescriptor}"/>, 
+        /// one for each <see cref="ViewComponent"/>.</returns>
+        public IEnumerable<TagHelperDescriptor> CreateDescriptors(string assemblyName)
+        {
+            if (assemblyName == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyName));
+            }
+
+            var viewComponentDescriptors = _descriptorProvider
+                .GetViewComponents()
+                .Where(viewComponent => string.Equals(assemblyName, viewComponent.TypeInfo.Assembly.GetName().Name,
+                    StringComparison.Ordinal));
+
+            var tagHelperDescriptors = CreateDescriptors(viewComponentDescriptors);
+            return tagHelperDescriptors;
+        }
+
+        private IEnumerable<TagHelperDescriptor> CreateDescriptors(
```

Comment:
Hm? no need for the method, nuke it and just have the .Select as a continuation of the linq statement in the calling method.